### PR TITLE
Added monthly cron job for units usage 

### PR DESCRIPTION
**Pull request by valyo**
A monthly cron job for units usage and a test for it which should be further developed

Please include the following in this section

- [x] Summary of the changes and the related issue
- [ ] Relevant motivation and context
- [ ] Any dependencies that are required for this change

Fixes # DDS-1285

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

Please delete options that are not relevant.

- [ ] Any dependent changes have been merged and published in downstream modules
- [x] Rebase/merge the branch which this PR is made to

## Formatting and documentation

- [x] I have added a row in the [changelog](../CHANGELOG.md)
- [x] The code follows the style guidelines of this project: Black / Prettier formatting
- [ ] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

## Tests

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

<a href="https://gitpod.io/#https://github.com/ScilifelabDataCentre/dds_web/pull/1181"></a>

### CODE REVIEW

File: dds_web/scheduled_tasks.py [hunk 1]
```diff
@@ -212,3 +212,35 @@ def delete_invite():
...
+
+    scheduler.app.logger.debug("Task: Collecting monthly usage information from Safespring.")
+    import sqlalchemy
+
+    from dds_web import db
+    from dds_web.database import models
+
+    # a mock dict with data that should be obtained from Safesprig's API
+    safespring_data = {
+        "safespring_name_1": {
```
Comment (inaod568) [inline dds_web/scheduled_tasks.py:230]: Exchange `safespring_name_1` and `safespring_name_2` with the `safespring_name` in the database for the existing unit(s?). Maybe do something like 
```
safespring_data = {}
for unit in db.session.query(models.Unit).all():
    safespring_data[unit.safespring_name] = {
            "TotalBytes": 434595434499,
            "TotalBytesRounded": 1434614451200,
            "TotalEntries": 10333,
        }
```

File: dds_web/scheduled_tasks.py [hunk 2]
```diff
@@ -212,3 +212,35 @@ def delete_invite():
...
+            "TotalBytes": 434595434499,
+            "TotalBytesRounded": 1434614451200,
+            "TotalEntries": 10333,
+        },
+    }
+
+    with scheduler.app.app_context():
+
+        for safespring_project in safespring_data:
+            usage = f'Usage for project {safespring_project}: {safespring_data[safespring_project]["TotalBytes"]}'
```
Comment (inaod568) [inline dds_web/scheduled_tasks.py:245]: You could get the keys and values instead
```suggestion
        for safespring_project, usage_info in safespring_data.items():
            usage = f'Usage for project {safespring_project}: {usage_info["TotalBytes"]}'
```

**Review by inaod568 — CHANGES_REQUESTED** [review]

File: dds_web/scheduled_tasks.py (hunk 1 above)
Comment (valyo) [reply dds_web/scheduled_tasks.py:230]: of course, I was mixing up units with projects here :)

File: dds_web/scheduled_tasks.py [hunk 3]
```diff
@@ -212,3 +212,31 @@ def delete_invite():
...
+        safespring_data = {}
+        for unit in db.session.query(models.Unit).all():
+            safespring_data[unit.safespring_name] = {
+                "TotalBytes": 434595434499,
+                "TotalBytesRounded": 1434614451200,
+                "TotalEntries": 10333,
+            }
+
+        for safespring_project, usage_info in safespring_data.items():
+            usage = f'Total usage for unit {safespring_project}: {usage_info["TotalBytes"]}'
```
Comment (inaod568) [inline dds_web/scheduled_tasks.py:241]:
```suggestion
            usage = f"Total usage for unit {unit.name} ({safespring_project}): {usage_info['TotalBytes']}"
```

**Review by inaod568 — CHANGES_REQUESTED** [review]
One tiiny thing that is actually not important atm but still. Otherwise - looks good! 

The change from `'` to `"` for the external quotes is mostly a personal preference and for consistency within the DDS

**Review by inaod568 — APPROVED** [review]
Looks good!

**Pull request merged**